CA2022 error in RazorSdk tool source

New SDK introduces new warnings that turn into errors in VMR build - see https://github.com/dotnet/installer/pull/19145#issuecomment-2027668850

```
##[error]/vmr/src/sdk/src/RazorSdk/Tool/Client.cs(172,31): error CA2022: (NETCORE_ENGINEERING_TELEMETRY=Build) Avoid inexact read with 'System.IO.Stream.ReadAsync(byte[], int, int, System.Threading.CancellationToken)' (https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca2022)
##[error]/vmr/src/sdk/src/RazorSdk/Tool/ConnectionHost.cs(110,31): error CA2022: (NETCORE_ENGINEERING_TELEMETRY=Build) Avoid inexact read with 'System.IO.Stream.ReadAsync(byte[], int, int, System.Threading.CancellationToken)' (https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca2022)
```

I'm going to patch to disable this warning, in VMR builds. However this needs to be properly fixed.

--- a/src/RazorSdk/Tool/Client.cs
+++ b/src/RazorSdk/Tool/Client.cs
@@ -169,7 +169,9 @@ namespace Microsoft.NET.Sdk.Razor.Tool
                     try
                     {
                         ServerLogger.Log($"Before poking pipe {Identifier}.");
+#pragma warning disable CA2022 // Avoid inexact read
                         await Stream.ReadAsync(Array.Empty<byte>(), 0, 0, cancellationToken);
+#pragma warning restore CA2022
                         ServerLogger.Log($"After poking pipe {Identifier}.");
                     }
                     catch (OperationCanceledException)

--- a/src/RazorSdk/Tool/ConnectionHost.cs
+++ b/src/RazorSdk/Tool/ConnectionHost.cs
@@ -107,7 +107,9 @@ namespace Microsoft.NET.Sdk.Razor.Tool
                     try
                     {
                         ServerLogger.Log($"Before poking pipe {Identifier}.");
+#pragma warning disable CA2022 // Avoid inexact read
                         await Stream.ReadAsync(Array.Empty<byte>(), 0, 0, cancellationToken);
+#pragma warning restore CA2022
                         ServerLogger.Log($"After poking pipe {Identifier}.");
                     }
                     catch (OperationCanceledException)

